x-pack/platform/plugins/shared/fleet/server/services/agentless/deployment_sync.ts

```diff
@@ -0,0 +1,163 @@
+                  logger.error(
+                    `[Agentless Deployment Sync] Failed to delete deployment ${deployment.policy_id}`,
+                    { error }
+                  );
+                });
+            }
+          } else if (
+            deployment.revision_idx === undefined ||
+            deployment.revision_idx < agentPolicy.revision
+          ) {
+            logger.info(
+              `[Agentless Deployment Sync]${dryRunTag(opts?.dryRun)} Updating deployment ${
+                deployment.policy_id
+              }`
+            );
+            if (!opts?.dryRun) {
+              const spacedScoppedSoClient = appContextService.getInternalUserSOClientForSpaceId(
+                agentPolicy.namespace || undefined
+              );
+              const esClient = appContextService.getInternalUserESClient();
+              await agentlessAgentService
+                .createAgentlessAgent(esClient, spacedScoppedSoClient, agentPolicy)
+                .catch((error) => {
+                  logger.error(
+                    `[Agentless Deployment Sync] Failed to update deployment ${deployment.policy_id}`,
+                    { error }
+                  );
+                });
+            }
+          }
+        },
+        {
+          concurrency: AGENTLESS_CONCURRENCY,
+        }
+      );
+      currentDeployments.push(...deploymentRes.deployments);
+    }
+
+    const agentlessPolicies = await agentPolicyService.list(soClient, {
+      perPage: SO_SEARCH_LIMIT,
+      fields: ['revision', 'supports_agentless', 'global_data_tags'],
+      kuery: `${AGENT_POLICY_SAVED_OBJECT_TYPE}.supports_agentless:true`,
+      spaceId: '*',
+    });
+    await pMap(
+      agentlessPolicies.items,
+      async (agentPolicy) => {
+        const deployment = currentDeployments.find((d) => d.policy_id === agentPolicy.id);
+        if (!deployment) {
+          logger.info(
+            `[Agentless Deployment Sync]${dryRunTag(opts?.dryRun)} Creating deployment for policy ${
+              agentPolicy.id
+            }`
+          );
+          if (!opts?.dryRun) {
+            const spacedScoppedSoClient = appContextService.getInternalUserSOClientForSpaceId(
+              agentPolicy.namespace || undefined
+            );
+            const esClient = appContextService.getInternalUserESClient();
+            await agentlessAgentService
+              .createAgentlessAgent(esClient, spacedScoppedSoClient, agentPolicy)
+              .catch((error) => {
+                logger.error(
+                  `[Agentless Deployment Sync] Failed to create deployment ${agentPolicy.id}`,
+                  { error }
+                );
+              });
+          }
+        }
+      },
+      {
+        concurrency: AGENTLESS_CONCURRENCY,
+      }
+    );
+    logger.info(`[Agentless Deployment Sync] Finished sync process`);
+  } catch (error) {
+    logger.error(`[Agentless Deployment Sync] Sync failed with error`, { error });
```

Comment:
Q as i don't see it anywhere. if there is a mismatch in version and you're targeting API where GET is not yet supported to list deployments, do you differentiate this error? will this be handled the same way as let's say 500 processing error

Reply:
It's now handled and we will only log a warning in that case